custom_components/tahoma/sensor.py

```diff
@@ -134,6 +140,8 @@ def state(self):
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
+        if self.device_class in UNITS_BY_DEVICE_CLASS:
```

Comment:
I would prefer to have this after we check for `self.device.attributes`, since this should be leading. This would require a small addition to the if check below.

Reply:
I already tried that but the check for self.device.attributes will match my CO2Sensor because the CO2Sensor has attributes but not the good one "CORE_MEASURED_VALUE_TYPE".
Maybe the "if self.devices.attributes" could be more specific with a check that at least one attribute is named "CORE_MEASURED_VALUE_TYPE"?

Reply:
@tillstaff, sorry for the slow reply. I have been working with core to have a CO/CO2 device class added! :) 
https://github.com/home-assistant/core/pull/47487